iotileship/iotile/ship/recipe.py

```diff
@@ -0,0 +1,76 @@
+import os
+from iotile.core.exceptions import ArgumentError
+from iotile.ship.exceptions import *
+import actions
+
+class RecipeObject (object):
+    """
+    An object containing the pipeline for processing recipes
+    
+    An entire pipeline configuration can be serialized to/from a json file to allow
+    for easy setup/version control of pcb pipelines.
+    """
+
+    def __init__(self, args):
+        self._steps         = []
+        self._name          = args.get("name", None)
+        self._description   = args.get("description", None)
+
+    @classmethod
+    def CreateStep(cls, step_type):
+        """
+        Create a step by specifying its name
+        """
+        if cls._actions_dict.get(step_type, None) is not None:
+            return cls._actions_dict.get(step_type)
+
+        raise UnknownRecipeActionType("Unknown stage type specified, cannot create it", name=step_type)
+
+
+    @classmethod
+    def FromFile(cls, path, actions_dict, file_format="yaml"):
+        cls._actions_dict = actions_dict
+
+        format_map = {
+            "json": cls._process_json,
+            "yaml": cls._process_yaml,
```

Comment:
Do we really want/need to support two formats?

Reply:
Not a big deal from the coding stand point, but it creates room for one person doing one thing and another doing another, and for something like this, I am not sure there is a reason to allow for choice

Reply:
Good point, the preference is to support only yaml.